### PR TITLE
Clean split of opencog code vs python/ROS code

### PR DESCRIPTION
**Pull request by linas**
The long-term plan is to move the core behavior and environment model into the main opencog repo, in some robot-embodiment directory. Before this can be done, a clean split of ROS vs. opencog code is needed: and that is what this pull request provides.  Tested, seems to work; more testing would be good.

### CODE REVIEW

File: movement/do-movement.scm [hunk 1]
```diff
@@ -0,0 +1,205 @@
+;
+; do-movement.scm
+;
+; Implement the movement API for ROS/blender animations.
+;
+
+; Delete the current definition, if any.
+(define (delete-definition STR)
```
Comment (amebel) [inline movement/do-movement.scm:8]: Why is this needed?

File: movement/movement.scm [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import rosgraph
+# This hard-coded install path from CMakefile
+sys.path.append('/usr/local/share/opencog/python')
+try:
+    # Throw an exception if roscore is not running.
+    rosgraph.Master('/rostopic').getPid()
+    if (os.path.isfile('atomic.py')):
+        # Python3 does not support execfile any longer
+        # execfile('atomic.py'))
+        exec(open('atomic.py').read())
```
Comment (amebel) [inline movement/movement.scm:35]: Why switch from `execfile` to `exec(open(...`?

File: movement/do-movement.scm (hunk 1 above)
Comment (linas) [reply movement/do-movement.scm:8]: DefineLink definitions cannot ordinarily be re-defined.  However, here, we want to have two different definitions of these DefinedPredicateNodes: one version which talks to ROS, and one version which does not.  If the non-ROS version is loaded first, and then the ROS version is loaded later, the first set of DefineLinks needs to be deleted.

There are other ways of solving this problem, too. This seemed like ... one of the better ones.

Another solution would be to provide two python versions: one with and one without ROS (i.e. by duck-typing the python code, instead of duck-typing the DefinedPredicateLinks)  This had the disadvantage of forcing the use of python; instead, it seemed like the duck-typed DefinedPredicateLinks would be better, as maybe some future version could use ... c++, haskell, or atomese, or whatever, to define these.

File: movement/movement.scm (hunk 2 above)
Comment (linas) [reply movement/movement.scm:35]: python3 does not support execfile ...

Comment (linas) [reply movement/movement.scm:35]: yes, ros is currently python2 only, but I figure that having code that works with either version would be better.
